scarb/src/ops/fmt.rs

```diff
@@ -92,6 +108,10 @@ fn print_diff(ws: &Workspace<'_>, path: &Path, diff: impl Display) {
         .print(format!("Diff in file {}:\n {}", path.display(), diff));
 }
 
+fn print_formatted_file(ws: &Workspace<'_>, path: &Path, file: impl Display) {
```

Comment:
not so sure about this

Reply:
One way to avoid this, would be to create some trait like `Emittable`, which would have an interface for emitting results of formatting. Then we could pass a zero sized struct implementing this method to `emit_formatted_file` instead of doing match at `159`.

We have done something similar in the compiler repo  https://github.com/starkware-libs/cairo/blob/f0c9a677fcea54f567c707f25f22a507560af5e4/crates/cairo-lang-formatter/src/cairo_formatter.rs#L87C3-L141

Reply:
Ok let me try this